### PR TITLE
[#123] Base backup functionality integration

### PR DESCRIPTION
**Pull request by Jubilee101**

### CODE REVIEW

**Comment by Jubilee101** [comment]
Hmm, the ci check failed. I'm working on it.

**Comment by Jubilee101** [comment]
I found that postgres sometimes sends me copydata message with no actual payload (what the hell?). So I added some extra checking to prevent false alarm.

File: src/libpgmoneta/wf_backup.c [hunk 1]
```diff
@@ -28,17 +28,21 @@
 
 /* pgmoneta */
 #include <pgmoneta.h>
+#include <security.h>
```
Comment (jesperpedersen) [inline src/libpgmoneta/wf_backup.c:31]: Sorting

File: src/libpgmoneta/wf_backup.c [hunk 2]
```diff
@@ -28,17 +28,21 @@
...
 #include <message.h>
-#include <network.h>
-#include <security.h>
 #include <tablespace.h>
 #include <utils.h>
 #include <workflow.h>
 
 /* system */
 #include <stdlib.h>
+#include <string.h>
```
Comment (jesperpedersen) [inline src/libpgmoneta/wf_backup.c:44]: Sorting

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgmoneta/wf_backup.c [hunk 3]
```diff
@@ -27,18 +27,22 @@
  */
 
 /* pgmoneta */
-#include <pgmoneta.h>
 #include <info.h>
 #include <logging.h>
+#include <memory.h>
 #include <message.h>
 #include <network.h>
+#include <pgmoneta.h>
```
Comment (jesperpedersen) [inline src/libpgmoneta/wf_backup.c:35]: This one is always first

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgmoneta/wf_backup.c (hunk 3 above)
Comment (Jubilee101) [reply src/libpgmoneta/wf_backup.c]: 😆 So sorry about that. It's fixed now.

**Comment by jesperpedersen** [comment]
Rebased, and merged.

Thanks for your contribution !

**Pull request merged**